### PR TITLE
Blog App: Add API endpoints

### PR DESCRIPTION
**Pull request by Timbar09**
In this PR, we ( @Timbar09 and @eudondian ) added API endpoints.

**Changes**
- Created an API endpoint to list all posts for a user.
- Created an API endpoint to list all comments for a user's post.
- Create an API endpoint to add a comment to a post.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
   resources :users, only: [:index, :show] do
     resources :posts, only: [:index, :show, :new, :create, :destroy] do
-      resources :comments, only: [:new, :create, :destroy]
+      resources :comments, only: [:index, :new, :create, :destroy]
       resources :likes, only: [:create]
     end
```
Comment (Zuhaib042) [inline config/routes.rb]: - [ ] Kindly consider adding routes for api. right now the routes you have is the normal routes. to add the api endpoints, you need to add the api endpoints.
consider checking this [resource out](https://medium.com/@ryanwaldorf/how-to-add-an-api-to-an-existing-rails-app-1d2419ba91bd)

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -1,6 +1,11 @@
 class CommentsController < ApplicationController
   before_action :authenticate_user!
-  before_action :set_post_and_user, only: %i[new create destroy]
+  before_action :set_post_and_user, only: %i[index new create destroy]
+
+  def index
+    @comments = @post.comments
+    render json: @comments
+  end
```
Comment (Zuhaib042) [inline app/controllers/comments_controller.rb:8]: - [ ] Kindly consider adding API controllers for your API endpoints. you can take help from this [resource](https://medium.com/@ryanwaldorf/how-to-add-an-api-to-an-existing-rails-app-1d2419ba91bd). I could not get json data from postman from api endpoints.

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi @Timbar09 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✅ Well done try to add api endpoints

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by bobb-Rob — APPROVED** [review]
## STATUS: APPROVED ✔️ 🍾 🍾 🍾 🥳 🥳 🥳 

Hi @Timbar09 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ All linters checks are ok
✅ The correct workflow is used. 
✅ Very descriptive Pull Request(PR)
✅ Your work is well documented.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Timbar09** [comment]
Thank you for the feedback, @bobb-Rob and @Zuhaib042.

**Pull request merged**